core/src/main/scala/chisel3/Aggregate.scala

```diff
@@ -503,6 +502,7 @@ abstract class Record(private[chisel3] implicit val compileOptions: CompileOptio
           case _ => ActualDirection.Bidirectional(ActualDirection.Default)
         }
     }
+    setElementRefs()
```

Comment:
It should be equivalent, but the above could be put in a `finally` block. (Not sure if that's clearer in any way or there's any real benefit to do ing so.)

Reply:
I thought about it, IMO think `finally` implies that the `catch` block rethrows (which it doesn't), but I can see the argument.